### ISSUE TITLE
Fully Convolutional Nets with varying input sizes - able to train batches?

### ISSUE THREAD

**Issue by 9thDimension** [fchollet/keras#3434]
Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by 9thDimension** [comment]
Since fully convolutional nets doing pixel-wise prediction accept inputs/outputs of varying length, how should batches of data be handled, given that the shape of the batch must be known in advance?

Padding with zeros is a workaround, but introduces vast areas of blank space mapping to an artificially constructed "blank space" output category. Less than ideal.

Can masking be used in this context?
